### ISSUE TITLE
feat: message -> regex -> form

### ISSUE THREAD

**Issue by martinhoess** [bendera/vscode-commit-message-editor#100]
Nice extension! 
There is only one thing I "miss":

It would be great if you could define a regex that breaks down the current message and fills the form with the values from the regex.

This would also allow you to display the n most recent commits on the form page and, when selected, run it through the regex and fill the form.

so for my case something like this
[https://regex101.com/r/Y5LNaQ/2](https://regex101.com/r/Y5LNaQ/2)
![image](https://github.com/bendera/vscode-commit-message-editor/assets/47973549/d9518f9f-4bce-413d-8869-4da11555cb9f)

**Comment by bendera** [comment]
Thank you for your feedback!

This would be a cool feature and actually, I was thinking about it. However, I have no idea how could be implemented. Since a commit message is plain text, not a programming language it's impossible to distinguish the values in the generated text. Or at least my knowledge is not enough for that. For example, the body of the message is a multi-line text, it can contain multiple paragraphs. How could be isolated from other paragraphs?

**Comment by martinhoess** [comment]
This is something that the person who defines the RegEx has to take care of himself. The only thing you can do is to assign the named groups from the regex result to the defined tokens.

So like in my example there is a named group "TYPE" (defined by the `?<TYPE>`) and so then there must be a token "type" in the extension config. if there is no token the named group will be ignored and if there is no named group the token will behave as if it was defined normally.

I have no idea how the VSCode extensions work (internally). But I think you "just" need to run the regex on the message and then map the result of the regex (the named groups) to the token.

**Comment by bendera** [comment]
So, if I understand correctly, the person who defines the form configuration could assign a "restore pattern" regexp to every token. It would make sense.

**Comment by martinhoess** [comment]
Yes. The more strictly the message is defined, the easier it is to define the regular expression for it.
The multi-line part is of course difficult or impossible, depending on whether there is a logical way to split it up. Also, there will always be cases where something is wrong or not recognised.

For example, if instead of `fix my special service: ...` I write `fixing my special service: ...` then fix is still recognized as "type". This could be solved in the expression by saying that a space must necessarily follow the type. 

But I think for the simple/short commit messages that (in my experience) most people/projects use, this would help/work.

https://regex101.com/r/I67Ss6/1
![image](https://github.com/bendera/vscode-commit-message-editor/assets/47973549/04d44b94-cbe5-48bd-a675-f375628bed68)

**Comment by evenkov** [comment]
Hi I found this perfect extension and really like it! However I agree with the idea to has option for RegEx validations. I imagine it like this:

1. **As Form**  - when You define `token` and it is **text** type to be able to define regular expression that will check this input. We use Axosoft - Scrum Agile Project Management and in our commit there is tag to Axosoft item. So then I define token {AxoItem} I prefer to validate it like this `[f|d]:\d{4,6}`. And after I input value for this token it will validate and coloured if does not match.
2. **As Text** - when define simple template to use a global RegEx for validate all commit message.
Now new use GitExtensions where there is option to validate commit message with RegEx. We used now this RegEx for validation. Here is:

[Git commit validation](https://regex101.com/r/jBX7Tx/1)


So it would be nice to have such validation and this cool extension!

**Comment by bendera** [comment]
Okay, I'm convinced.
